Utils/azuretools-core/src/com/microsoft/azuretools/adauth/AuthContext.java

```diff
@@ -238,7 +238,11 @@ protected AuthResult getTokenWithRefreshToken(@NotNull final String refreshToken
                 Map<String, String> requestParameters = new HashMap<>();
                 if (!StringUtils.isNullOrEmpty(resource)) {
                     requestParameters.put(OAuthParameter.Resource, resource);
+                    if (resource.equals(OAuthParameter.EspTokenResource)) {
```

Comment:
You could add a new polymorphism method to the abstract class `CacheDriver` with `scope` parameter.

Reply:
I have tried this but find out a lot of methods need to modify. I think we can delay the code refactor to next time.

---
In reply to: [336864978](https://github.com/microsoft/azure-tools-for-java/pull/3473#discussion_r336864978) [](ancestors = 336864978)